src/common/options/osd.yaml.in

```diff
@@ -27,6 +27,16 @@ options:
   - osd_numa_auto_affinity
   flags:
   - startup
+- name: set_keepcaps
+  type: bool
```

Comment:
I think this option is redundant.
If we need some capabilities, then user "conf.setuser" should have this capabilities.
Configuring "conf.setuser" seems like operator-related task.

Inheriting all capabilities from starting user (usually root) seems like serious overkill and effectively circumvents access control.

Reply:
The keep caps is intended to work in conjunction with the capability trimming of the plugin loader. Changing the permitted capability set is only subtractive, it is impossible to add capabilities later on. The plugin system trims down the capabilities to the merged requirements of all plugins. The VDO plugin requires no capability, so the plugin system removes all capabilities if no plugin or only the VDO plugin is loaded. The set_keepcaps option serves as additional method to remove all capabilities from the start if the user so choses.

Reply:
Let me rephrase this:
The default way of starting ceph is to have the initialization run with root privileges and then switch via setuid to the ceph user to have ceph operate in a more restricted and secure environment. The setuid will always cause all capabilities to be lost unless KEEPCAPS is set. There is no way to reestablish capabilities within a process once they are removed. 
It is different if ceph would be started from the get go under the ceph userid, then applying capabilities to the ceph executable via file system attributes is possible.
The approach taken in this patch is as follows:
1) KEEPCAPS is only set if a specific option is activated. This option is off by default, so no installation will have any capabilities available unless this option is explicitly set.
2) In addition to 1), the plugin system will remove all capabilities unless they are explicitly requested by plugins loaded at startup time. The default set of plugins (VDO) does not request any capabilities, so 2) will also remove all capabilities by default.
Only if additional plugins are added, and one of the plugins requests a capability, and the ceph system option is explicitly set to set KEEPCAPS, then the capability requested will be in the permitted set.
At this point, it is not yet even possible to invoke system calls that require the capability. Capability-aware code first needs to add the capability present in the permitted set to the effective set and then do the system call. Plugins we have written do that right before the critical sys call and then wipe the effective set immediately after, keeping the rest of the ceph code in a more contained state.

Reply:
@mohmacht 
Thank you for the explanation.
I needed to read much deeper then before into linux capabilites, and understand difference between permitted, effective and ambient to understand that my initial opinion on "add permission to user" was just incoherent with reality.